autorest/azure/async.go

```diff
@@ -81,10 +81,37 @@ func (f *Future) Done(sender autorest.Sender) (bool, error) {
 
 	resp, err := sender.Do(f.req)
 	f.resp = resp
-	if err != nil || !autorest.ResponseHasStatusCode(resp, pollingCodes[:]...) {
+	if err != nil {
 		return false, err
```

Comment:
so here previously we would just return nil for `err` in the case the error was in the body. when would it be one or the other ? ( i.e. err != nil or the error in the body).

Reply:
sender.Do() would return a non-nil error e.g. cases of transport-level failure (say the network connection is lost) and I'm sure there are other cases (I would think of this case where the HTTP client fails to send anything over the wire).